pkg/openservicebroker/api/types.go

```diff
@@ -0,0 +1,160 @@
+
+type LastOperationState string
+
+const (
+	LastOperationStateInProgress LastOperationState = "in progress"
+	LastOperationStateSucceeded  LastOperationState = "succeeded"
+	LastOperationStateFailed     LastOperationState = "failed"
+)
+
+type ProvisionRequest struct {
+	ServiceID         string            `json:"service_id"`
+	PlanID            string            `json:"plan_id"`
+	Parameters        map[string]string `json:"parameters,omitempty"`
+	AcceptsIncomplete bool              `json:"accepts_incomplete,omitempty"`
+	OrganizationID    string            `json:"organization_guid"`
+	SpaceID           string            `json:"space_guid"`
+}
+
+type ProvisionResponse struct {
+	DashboardURL string    `json:"dashboard_url,omitempty"`
+	Operation    Operation `json:"operation,omitempty"`
+}
+
+type Operation string
+
+type UpdateRequest struct {
+	ServiceID         string            `json:"service_id"`
+	PlanID            string            `json:"plan_id,omitempty"`
+	Parameters        map[string]string `json:"parameters,omitempty"`
+	AcceptsIncomplete bool              `json:"accepts_incomplete,omitempty"`
+	PreviousValues    struct {
+		ServiceID      string `json:"service_id,omitempty"`
+		PlanID         string `json:"plan_id,omitempty"`
+		OrganizationID string `json:"organization_id,omitempty"`
+		SpaceID        string `json:"space_id,omitempty"`
+	} `json:"previous_values,omitempty"`
+}
+
+type UpdateResponse struct {
+	Operation Operation `json:"operation,omitempty"`
+}
+
+type BindRequest struct {
+	ServiceID    string `json:"service_id"`
+	PlanID       string `json:"plan_id"`
+	AppGUID      string `json:"app_guid,omitempty"`
+	BindResource struct {
+		AppGUID string `json:"app_guid,omitempty"`
+		Route   string `json:"route,omitempty"`
+	} `json:"bind_resource,omitempty"`
+	Parameters map[string]string `json:"parameters,omitempty"`
+}
+
+type BindResponse struct {
+	Credentials     map[string]interface{} `json:"credentials,omitempty"`
+	SyslogDrainURL  string                 `json:"syslog_drain_url,omitempty"`
+	RouteServiceURL string                 `json:"route_service_url,omitempty"`
+	VolumeMounts    []interface{}          `json:"volume_mounts,omitempty"`
+}
+
+type UnbindResponse struct {
+}
+
+type DeprovisionResponse struct {
+	Operation Operation `json:"operation,omitempty"`
+}
+
+type ErrorResponse struct {
+	Description string `json:"description"`
+}
+
+// asyncRequiredResponse type is not formally defined in the spec
+type asyncRequiredResponse struct {
+	Error       string `json:"error,omitempty"`
+	Description string `json:"description"`
+}
+
+var AsyncRequired = asyncRequiredResponse{
+	Error:       "AsyncRequired",
+	Description: "This service plan requires client support for asynchronous service operations.",
+}
+
+// from http://docs.cloudfoundry.org/services/catalog-metadata.html#services-metadata-fields
+
+const (
+	ServiceMetadataDisplayName         = "displayName"
+	ServiceMetadataImageURL            = "imageUrl"
```

Comment:
OK; as we're adding what is arguably a vendor extension here, I suppose that on balance, on the service broker API side, we'd be better calling it something like "template.openshift.io/iconClass" rather than "iconClass" ?  (Clearly the name "iconClass" is already fixed on the OpenShift API side).